ref/binding.go

```diff
@@ -0,0 +1,139 @@
+//go:build cgo && ((linux && amd64) || (darwin && amd64) || (darwin && arm64) || (windows && amd64)) && (sphincs_haraka_128f || sphincs_haraka_128s || sphincs_haraka_192f || sphincs_haraka_192s || sphincs_haraka_256f || sphincs_haraka_256s || sphincs_sha2_128f || sphincs_sha2_128s || sphincs_sha2_192f || sphincs_sha2_192s || sphincs_sha2_256f || sphincs_sha2_256s || sphincs_shake_128f || sphincs_shake_128s || sphincs_shake_192f || sphincs_shake_192s || sphincs_shake_256f || sphincs_shake_256s)
+
+package sphincsplus
+
+//#cgo amd64 LDFLAGS: -L./
+//#cgo amd64 CFLAGS: -O3 -std=c99 -D CGO=1
+//#cgo darwin/amd64 LDFLAGS: -L./
+//#cgo darwin/amd64 CFLAGS: -O3 -std=c99 -D CGO=1
+//#cgo linux/amd64 LDFLAGS: -L./ -L/usr/lib/x86_64-linux-gnu/
+//#cgo linux/amd64 CFLAGS: -O3 -std=c99 -D CGO=1
+//#include "api.h"
+import "C"
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/katzenpost/sphincsplus/ref/params"
+)
+
+var (
+
+	// PublicKeySize is the size in bytes of the public key.
+	PublicKeySize int = C.CRYPTO_PUBLICKEYBYTES
+
+	// PrivateKeySize is the size in bytes of the private key.
+	PrivateKeySize int = C.CRYPTO_SECRETKEYBYTES
+
+	// SignatureSize is the size in bytes of the signature.
+	SignatureSize int = C.CRYPTO_BYTES
+
+	// SignatureName is the parameterized signature system name
+
+	// Name returns the string naming of the current
+	// Sphincs+ that this binding is being used with.
+	SignatureName = params.Name()
```

Comment:
I'd say it's fine if it works... this variable assignment. Sometimes golang programmers will do that inside an init function but i'm not clear on when that's necessary.